src/controllers/bulk/bulkcontroller.h

```diff
@@ -43,6 +43,9 @@ class BulkController : public Controller {
 
     virtual std::shared_ptr<LegacyControllerMapping> cloneMapping() override;
     void setMapping(std::shared_ptr<LegacyControllerMapping> pMapping) override;
+    void clearMapping() override {
+        m_pMapping = nullptr;
+    }
```

Comment:
just call `setMapping` with a `nullptr` directly.

Reply:
That was the first thing I tried.
Gives me this when selecting No mapping
```
DEBUG ASSERT: "pMapping.use_count() == 1" in function std::shared_ptr<_Tp> Controller::downcastAndTakeOwnership(std::shared_ptr<LegacyControllerMapping>&&) [with SpecificMappingType = LegacyMidiControllerMapping] at ./src/controllers/controller.h:89
```

Reply:
https://github.com/mixxxdj/mixxx/blob/815219d8a048165afd60c5283874da71f6f3e342/src/controllers/controller.h#L82-L96

Reply:
right, that function does not take that usecase into account. Simply add a `if (pMapping == nullptr) { return nullptr; }` so it supports that usecase. Alternatively you can add the appropriate handling to the `setMapping` implementations.

Reply:
Okay. I only took a brief look at this and didn't want to touch it.
But yeah, that's a straight forward solution.
Thanks!